tests/unit/test_utils.py

```diff
@@ -646,3 +646,26 @@ def test_call_subprocess_closes_stdin():
 def test_remove_auth_from_url(auth_url, expected_url):
     url = remove_auth_from_url(auth_url)
     assert url == expected_url
+
+
+@pytest.mark.parametrize('auth_url, expected_url', [
```

Comment:
Maybe add test cases of:
1. containing `:` but no `@` (e.g. `domain.tld:8080`)
2. password containing a `:`

Also, I would order the test cases "simplest" to least simple, e.g. start with the test cases having no user-pass, and put all cases with password redaction at the end. The current ordering makes it harder to tell what cases are covered and left out.

Reply:
Does a password containing ":" need to be urlencoded?

Reply:
To expand the previous example:
```
>>> parsed = urlparse('https://user:pass:word@domain.tld/svn/project/trunk')
>>> parsed.password
'pass:word'
```
(But yes, it looks like it's supposed to be encoded.)

Reply:
> this is user input, it seems like we can't guarantee it will be encoded correctly, etc.

Yeah. This makes sense. @kvalev Could you add a case here: `'https://user:pass:word@domain.tld/svn/project/trunk'`?

Reply:
Already did, see the latest commit.

Reply:
Indeed. I was viewing an outdated diff. Sorry for the noise. :)